hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusQueueProcessors.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.command.bus;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.concurrent.ExecutorService;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import javax.inject.Inject;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.persistence.PostRemove;
+import javax.websocket.Session;
+
+import org.hawkular.bus.common.BasicMessage;
+import org.hawkular.bus.common.BasicMessageWithExtraData;
+import org.hawkular.bus.common.ConnectionContextFactory;
+import org.hawkular.bus.common.MessageProcessor;
+import org.hawkular.bus.common.consumer.ConsumerConnectionContext;
+import org.hawkular.cmdgw.Constants;
+import org.hawkular.cmdgw.GatewayLoggers;
+import org.hawkular.cmdgw.MsgLogger;
+import org.hawkular.cmdgw.annotations.Bus;
+import org.hawkular.cmdgw.annotations.Feed;
+import org.hawkular.cmdgw.command.Command;
+import org.hawkular.cmdgw.command.CommandContext;
+import org.hawkular.cmdgw.command.CommandContextFactory;
+import org.hawkular.cmdgw.command.Commands;
+import org.hawkular.cmdgw.command.ws.server.ConnectedSessions;
+import org.hawkular.cmdgw.command.ws.server.WebSocketHelper;
+
+@Startup
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
+public class BusQueueProcessors {
+
+    private static class FeedQueueListener<T extends BasicMessage>
+            extends org.hawkular.bus.common.consumer.AbstractBasicMessageListener<T> {
+        private final ConnectedSessions connectedFeeds;
+
+        public FeedQueueListener(ConnectedSessions connectedFeeds) {
+            super();
+            this.connectedFeeds = connectedFeeds;
+        }
+
+        @Override
+        public void onMessage(Message message) {
```

Comment:
> AbstractBasicMessageListener has its own onMessage()

You certainly mean `BasicMessageListener` that has its `onMessage()` and `onBasicMessage()`. I could not use those because they take the BasicMessage class to deserialize into from type argument `T extends BasicMessage` of the listener, which in turn assumes that listeners are implemented one per message class, which is not the case in my proposal. I rather have one listener for all message classes, that dynamically selects the command to handle the given message. But of course, to deserialize the message, I need another source of the class to deserialize into - hence I introduced a dedicated JMS message header to store that info: https://github.com/hawkular/hawkular-command-gateway/pull/14#discussion-diff-40366092R74